packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,193 @@
+import * as React from 'react';
+import { KeyCodes } from '../../Utilities';
+import { TimeConstants, addMinutes, formatTimeString, ceilMinuteToIncrement } from '@fluentui/date-time-utilities';
+import { ComboBox, IComboBox, IComboBoxOption } from '../../ComboBox';
+import { ITimePickerProps, ITimeRange, ITimePickerStrings } from './TimePicker.types';
+
+const REGEX_SHOW_SECONDS_HOUR_12 = /((1[0-2]|0?[1-9]):([0-5][0-9]):(?:[0-5]\d) ?([AaPp][Mm]))$/;
+const REGEX_HIDE_SECONDS_HOUR_12 = /((1[0-2]|0?[1-9]):([0-5][0-9]) ?([AaPp][Mm]))$/;
+const REGEX_SHOW_SECONDS_HOUR_24 = /([0-9]|0[0-9]|1[0-9]|2[0-3]):(?:[0-5]\d):(?:[0-5]\d)$/;
+const REGEX_HIDE_SECONDS_HOUR_24 = /([0-9]|0[0-9]|1[0-9]|2[0-3]):(?:[0-5]\d)$/;
+
+const TIME_LOWER_BOUND = 0;
+const TIME_UPPER_BOUND = 23;
+
+const getDefaultStrings = (useHour12: boolean, showSeconds: boolean): ITimePickerStrings => {
+  let errorMessageToDisplay = '';
+  const hourUnits = useHour12 ? '12-hour' : '24-hour';
+  showSeconds
+    ? (errorMessageToDisplay = `TimePicker format must be valid and in the ${hourUnits} ` + `format hh:mm:ss A.`)
+    : (errorMessageToDisplay = `TimePicker format must be valid and in the ${hourUnits} ` + `format hh:mm A.`);
+
+  return {
+    invalidInputErrorMessage: errorMessageToDisplay,
+  };
+};
+
+export const TimePicker: React.FunctionComponent<ITimePickerProps> = ({
+  label,
+  increments = 30,
+  showSeconds = false,
+  allowFreeform = true,
+  useHour12 = false,
+  timeRange,
+  strings = getDefaultStrings(useHour12, showSeconds),
+  onFormatDate,
+  onValidateUserInput,
+  onChange,
+  ...rest
+}: ITimePickerProps) => {
+  const [selectedKey, setSelectedKey] = React.useState<string | number | undefined>('');
+  const [userText, setUserText] = React.useState<string>('');
+  const [errorMessage, setErrorMessage] = React.useState<string>('');
+
+  const optionsCount = getDropdownOptionsCount(increments, timeRange);
+
+  const timePickerOptions: IComboBoxOption[] = React.useMemo(() => {
+    const optionsList = Array(optionsCount);
+    for (let i = 0; i < optionsCount; i++) {
+      optionsList[i] = 0;
+    }
+    const defaultTime = generateDefaultTime(increments, timeRange);
+
+    return optionsList.map((_, index) => {
+      const option = addMinutes(defaultTime, increments * index);
+      option.setSeconds(0);
+      const optionText = onFormatDate ? onFormatDate(option) : formatTimeString(option, showSeconds, useHour12);
+      if (index === 0) {
+        setSelectedKey(optionText);
```

Comment:
You shouldn't call state setters while rendering (`useMemo` runs while rendering). Instead, do one of the following:
- Move the definition of `selectedKey` below `timePickerOptions` so that you can do `const [selectedKey, setSelectedKey] = React.useState(timePickerOptions[0])`
- OR, move the `setSelectedKey` call to a `useEffect` which runs when `timePickerOptions` changes

Reply:
Resolved by following the first bullet point of moving `selectedKey` below `timePickerOptions` and then `const [selectedKey, setSelectedKey] = React.useState(timePickerOptions[0])`.